### DISCUSSION TITLE
Support for new Traefik Rules

### DISCUSSION THREAD

**Discussion by H1ghBre4k3r** [k3s-io/k3s#7876] [Q&A]
**Is your feature request related to a problem? Please describe.**

Some of my services should only be accessible from within my VPN. Therefore, I would like to utilize the `ClientIP` rule (see the [rules section on their documentation page](https://doc.traefik.io/traefik/routing/routers/#rule)), but it is not included in the traefik setup/CRDs currently. Manually applying new traefik CRDs somehow did not work out for me. 

**Describe the solution you'd like**
It would be great to have the "new" CRDs included with K3s/traefik. 

**Describe alternatives you've considered**

I tried to manually apply [the traefik CRDs described on their website](https://doc.traefik.io/traefik/reference/dynamic-configuration/kubernetes-crd/#definitions), which, as mentioned, did not work out for me. 

**Additional context**

In my current setup I heavily rely on the builtin traefik setup for my services. The usages include several (more or less) complicated configurations. For some of them, I would like to make use of [some "new" traefik rules](https://doc.traefik.io/traefik/routing/routers/#rule).

**Comment by brandond** [comment]
You shouldn't need to have the "new" CRD, the existing Traefik CRDs have supported rules for ages. What version of K3s are you using, and how did you specify the rule in your router? What was the result?

**Reply by H1ghBre4k3r** [reply]
Ah, please excuse my mistake! I messed up something with my config (and identified a wrong source for my issues). The CRDs (and rules) actually work. 

The main idea was to limit access to certain services depending on the IP address of the client, for which I was about to use the `ClientIP` rule. I've mistaken the malfunctioning of my solution to be an issue with the CRDs, but it's rather an issue of the client ip not being populated/forwarded as I thought. Discussing a solution for this (especially due to the "complexity" of my setup - k3s behind firewall, using an nginx stream reverse proxy for access) should probably not be done here, since there are other discussions/issues already opened for that. 

I'm very sorry for the confusion. ^^'
